MovingBlocks/Terasology
# feat(ModuleManager): do not load all modules in the classpath by default

As we were reminded in #4375, putting Terasology modules on the classpath is usually _not_ what we want to do. Additionally, the [draft of ModuleManager for gestalt v7](https://github.com/MovingBlocks/Terasology/blob/f987e3b57c2db0f3d883c200e64befdb9d58b765/engine/src/main/java/org/terasology/engine/core/module/ModuleManagerImpl.java#L56) doesn't load classpath modules at all.

This PR disables running `loadModulesFromClasspath` _unless_  the environment variable `terasology_load_classpath_modules` is set to `true`.


### How to test

* No expected impact on the shipped game.
* _Might_ impact running the game in dev workspaces. I think it _shouldn't,_ but I could be surprised here.
* Impacts automated tests. Check
  - tests in this repo (mostly `:engine-test:test`, but run all tests while we're at it.)
  - tests in module repos
    - tests using MTEExtension (ModuleTestingEnvironment)
    - non-MTE unit tests


### Outstanding before merging

- [ ] fix MTE?
